### PR TITLE
feature: basic example app

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Basic modern example for using RNTL. The app is generated from RN CLI template so there is some iOS/Android boilerplate but the benefit here is that we work on a real RN app and not an empty shell of it.

I also plan to use it as a base for refresh of other example apps.

### Test plan

All checks and tests pass.

### CODE REVIEW

File: examples/basic/App.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      </SafeAreaView>
+    );
+  }
+
+  return (
+    <SafeAreaView>
+      <Home user={user} />
+    </SafeAreaView>
+  );
+};
```
Comment (thymikee) [inline examples/basic/App.tsx:22]: less code is less code
```suggestion
  return (
    <SafeAreaView>
      {user == null ? <LoginForm onLoginSuccess={setUser} /> : <Home user={user} />}
    </SafeAreaView>
  );
};
```

File: examples/basic/__tests__/App.test.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  fireEvent.changeText(screen.getByLabelText('Password'), 'admin1');
+
+  // Hint: we can use `getByText` to find our button by its text.
+  fireEvent.press(screen.getByText('Sign In'));
+
+  // Idiom: since pressing button triggers async operation we need to use `findBy` query to wait
+  // for the action to complete.
+  // Hint: subsequent queries do not need to use `findBy`, because they are used after the async action
+  // already finished
+  expect(await screen.findByText('Welcome admin!')).toBeTruthy();
```
Comment (thymikee) [inline examples/basic/__tests__/App.test.tsx:43]: There's no need to expect everything, right? This reads better as a scenario to me:
```suggestion
  await screen.findByText('Welcome admin!'));
```

**Review by thymikee — COMMENTED** [review]
Looks good! For the ease of maintenance I'd strongly consider not committing the "android" and "ios" folders (we could generate those with `init` on demand) or use Expo

File: examples/basic/__tests__/App.test.tsx (hunk 2 above)
Comment (mdjastrzebski) [reply examples/basic/__tests__/App.test.tsx:43]: Yeah thats a good question, I'used that in a similar pattern as we suggest using `expect(getByText()).toBeTruthy()` to clarify the intent, despite the fact that simple `getByText` would do the same.

**Comment by mdjastrzebski** [comment]
I've migrated the example to use Expo managed.

File: examples/basic/.prettierrc.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  arrowParens: 'avoid',
+  bracketSameLine: true,
+  bracketSpacing: true,
+  singleQuote: true,
+  trailingComma: 'all',
+  printWidth: 100,
+};
```
Comment (thymikee) [inline examples/basic/.prettierrc.js:8]: let's remove this and apply the same prettier settings as the rest of the repo :)

Comment (mdjastrzebski) [reply examples/basic/.prettierrc.js]: done

**Review by thymikee — APPROVED** [review]

**Pull request merged**